File: Baekjoon/heetae/로봇 프로젝트.py
import sys

input = sys.stdin.readline

while True:
    try:
        size = int(input()) * 10000000
        n = int(input())
        lego = [int(input()) for _ in range(n)]
        lego.sort()
        i, j = 0, n - 1
        while i < j:
            if lego[i] + lego[j] == size:
                print("yes %d %d" % (lego[i], lego[j]))
                break
            elif lego[i] + lego[j] > size:
                j -= 1
            elif lego[i] + lego[j] < size:
                i += 1
        else:
            print("danger")
    except:
        break

File: Baekjoon/hyeonjin/로봇_프로젝트.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        String str = null;
        while((str = br.readLine()) != null) {
            int x = Integer.parseInt(str);
            int n = Integer.valueOf(br.readLine());
            int[] arr = new int[n];

            for (int i = 0; i < n; i++) {
                arr[i] = Integer.valueOf(br.readLine());
            }

            Arrays.sort(arr);
            int start = 0;
            int end = n - 1;
            String result = "danger";
            x *= 10_000_000;
            while (start < end) {
                int sum = arr[start] + arr[end];

                if (sum == x) {
                    result = "yes" + " " + arr[start] + " " + arr[end];
                    break;
                } else if (sum > x) {
                    end--;
                } else {
                    start++;
                }
            }

            System.out.println(result);
            str = null;
        }
    }
}
